### ISSUE TITLE
number() on a number returns null

### ISSUE THREAD

**Issue by LockeBirdsey** [camunda/feel-scala#595]
**Describe the bug**
Due to unexpected value types in a retrieved value, we sanitise types by `number(value)`. When `value` is already a number, `number(value)` returns `null`

**To Reproduce**
1. `number("17")` will return `17`
2. `number(17)` will return `null`

**Expected behavior**
`number(value)` should return a numeric representation of `value` unless there is no possible way for the cast to occur e.g. it is not a primitive.

To be fair, it makes sense that it returns null in this case and that we could further prevent against this issue by doing `number(string(value))`. Additionally the Zeebe docs do seem to say the input needs to be a string, so I'm not entirely certain if what I'm reporting is entirely valid or not.

**Environment**
* Zeebe 8.1.4

**Comment by saig0** [comment]
@LockeBirdsey thank you for raising this up. :+1: 

First of all, I understand your expectation. 

But the current behavior matches the [documentation](https://camunda.github.io/feel-scala/docs/reference/builtin-functions/feel-built-in-functions-conversion#number) since the function expects a string argument. 

The behavior is also aligned with the DMN specification (chapter 10.3.4.1, page 138):

![image](https://user-images.githubusercontent.com/4305769/221827869-3703fe78-187d-4113-8980-dc5527187561.png)

The `number()` function is only defined for string arguments. As a result, the function should return `null` for other arguments.

---

To sum it up, it is the **expected behavior** from the DMN perspective. So, we can't treat the behavior as a bug.

**Comment by saig0** [comment]
@LockeBirdsey do you understand the reasoning? Can you agree with it?

Feel free to share your example. I could tweak the expression with you.

**Comment by LockeBirdsey** [comment]
Ahh yes that makes complete sense, thanks for the info!

I'll handle my problem in a different way then.
